src/HotChocolate/Core/src/Abstractions/IExecutable.cs

```diff
@@ -1,12 +1,53 @@
+using System.Collections;
 using System.Threading;
 using System.Threading.Tasks;
 
+#nullable enable
+
 namespace HotChocolate
 {
+    /// <summary>
+    /// Represents a abstract executable that is well known in the framework. If the execution
+    /// engine encounters a <see cref="IExecutable"/>, it will call execute it
+    /// </summary>
     public interface IExecutable
     {
-        ValueTask<object> ExecuteAsync(CancellationToken cancellationToken);
+        /// <summary>
+        /// The current state of the executable
+        /// </summary>
+        object Source { get; }
```

Comment:
Do we need an abstract source? or is it enough to have it on the specific implementations?

Reply:
I guess it makes sense to have it there. We use it in Cursor Paging and Offset Paging